### PR TITLE
Replace latin-1 by latin1 in charset detection

### PR DESCRIPTION
**Pull request by MrPetovan**
See https://github.com/friendica/friendica/issues/5470#issuecomment-418351211

Part of #5470

### CODE REVIEW

File: src/Util/ParseUrl.php [hunk 1]
```diff
@@ -181,12 +181,12 @@ public static function getSiteinfo($url, $no_guessing = false, $do_oembed = true
...
-		if ($charset == '') {
-			$charset = 'utf-8';
-		}
-
-		if (($charset != '') && (strtoupper($charset) != 'UTF-8')) {
+		if ($charset && strtoupper($charset) != 'UTF-8') {
 			logger('detected charset ' . $charset, LOGGER_DEBUG);
+
+			// See https://github.com/friendica/friendica/issues/5470#issuecomment-418351211
+			$charset = str_replace('latin-1', 'latin1', $charset);
```
Comment (annando) [inline src/Util/ParseUrl.php]: You should possibly use ```str_ireplace```.

Comment (MrPetovan) [reply src/Util/ParseUrl.php]: Good idea, I'll use it when you provide me a more complete list of charset errors.

**Comment by MrPetovan** [comment]
Closed until I receive a bigger list of common charset errors.

**Comment by MrPetovan** [comment]
If not then I'll re-submit this one unchanged.

**Pull request closed without merging**